app/uk/gov/hmrc/eusubsidycompliancefrontend/services/AuditService.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eusubsidycompliancefrontend.services
+
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.Logging
+import play.api.libs.json.{Json, Writes}
+import play.api.mvc.Request
+import uk.gov.hmrc.eusubsidycompliancefrontend.models.AuditEvent
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.AuditExtensions.auditHeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.model.ExtendedDataEvent
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+
+@ImplementedBy(classOf[AuditServiceImpl])
```

Comment:
Do we need a separate trait? I was thinking of consolidating the `Foo` and `FooImpl` classes into a single `Foo` class, if we can, so things are a little more idiomatic.

Reply:
It's easy for me to write the test cases , i prefer tis way